### PR TITLE
Fixes a bug where user sees error message telling them to accept conditions of sale after they've already checked the checkbox

### PR DESCRIPTION
**Pull request by yuki24**
addresses https://artsyproduct.atlassian.net/browse/AUCT-1018

This fixes a bug where user sees error message telling them to accept conditions of sale after they've already checked the checkbox. The root cause of this seems to be a racing condition in `setFieldTouched` and `setFieldValue`, and flipping the function order fixed the issue, but I don't have a definite answer as to why it fixed it or the previous order caused a bug. There might be more places that have a similar bug.

I tried to write a test for this but sadly it didn't fail in test, so just leaving it here as a reference:

```tsx
it("validates against an unchecked conditions of sale checkbox", async () => {
  const env = setupTestEnv()
  const page = await env.buildPage()

  setupCreateTokenMock()
  await page.submitForm()

  expect(page.text()).toMatch("You must agree to the Conditions of Sale")

  page.agreeToTermsInput.props().onSelect(true)
  await page.update()

  expect(page.text()).not.toMatch("You must agree to the Conditions of Sale")
})
```

### Register

AUCT-1018-2

### Register and Bid

AUCT-1018

### CODE REVIEW

File: src/v2/Apps/Auction/Components/BidForm.tsx [hunk 1]
```diff
@@ -46,11 +46,9 @@ export interface FormValues {
...
 }
 
-Yup.addMethod(Yup.string, "present", function(message) {
-  return this.test("test-present", message, value => {
-    return this.trim()
-      .required(message)
-      .isValid(value)
+Yup.addMethod(Yup.string, "present", function (message) {
+  return this.test("test-present", message, (value) => {
+    return this.trim().required(message).isValid(value)
```
Comment (yuki24) [inline src/v2/Apps/Auction/Components/BidForm.tsx:51]: Most of these unrelated prettier changes and could be ignored.

File: src/v2/Apps/Auction/Components/BidForm.tsx [hunk 2]
```diff
@@ -252,35 +252,26 @@ export const BidForm: React.FC<Props> = ({
...
 
-                      <Box mx="auto" mb={3}>
+                      <Box mx="auto">
                         <ConditionsOfSaleCheckbox
                           selected={values.agreeToTerms}
-                          onSelect={value => {
-                            setFieldValue("agreeToTerms", value)
+                          onSelect={(value) => {
                             setFieldTouched("agreeToTerms")
+                            setFieldValue("agreeToTerms", value)
```
Comment (yuki24) [inline src/v2/Apps/Auction/Components/BidForm.tsx:265]: This part is the actual fix.

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  Danger failed to run `dangerfile.ts`.
  </td>
    </tr>
  </tbody>
</table>



## Error RangeError
```
Maximum call stack size exceeded
RangeError: Maximum call stack size exceeded
    at RegExp.[Symbol.replace] (<anonymous>)
    at String.replace (<anonymous>)
    at standardizeName (/home/circleci/project/node_modules/@babel/core/lib/config/files/plugins.js:94:15)
    at resolveStandardizedName (/home/circleci/project/node_modules/@babel/core/lib/config/files/plugins.js:98:28)
    at resolvePlugin (/home/circleci/project/node_modules/@babel/core/lib/config/files/plugins.js:54:10)
    at loadPlugin (/home/circleci/project/node_modules/@babel/core/lib/config/files/plugins.js:62:20)
    at createDescriptor (/home/circleci/project/node_modules/@babel/core/lib/config/config-descriptors.js:154:9)
    at /home/circleci/project/node_modules/@babel/core/lib/config/config-descriptors.js:109:50
    at Array.map (<anonymous>)
    at createDescriptors (/home/circleci/project/node_modules/@babel/core/lib/config/config-descriptors.js:109:29)
```
### Dangerfile
```
-----------------^
```
  
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against e7e7fd4a6c15a9c9a5dbb6f2b2a23de15793a1d8
</p>

File: src/v2/Apps/Auction/Components/RegistrationForm.tsx [hunk 3]
```diff
@@ -90,9 +90,9 @@ const InnerForm: React.FC<InnerFormProps> = props => {
         <Box mx="auto">
           <ConditionsOfSaleCheckbox
             selected={values.agreeToTerms}
-            onSelect={value => {
-              setFieldValue("agreeToTerms", value)
+            onSelect={(value) => {
               setFieldTouched("agreeToTerms")
+              setFieldValue("agreeToTerms", value)
```
Comment (yuki24) [inline src/v2/Apps/Auction/Components/RegistrationForm.tsx:95]: And this is the other fix.

**Comment by yuki24** [comment]
The CI is a bit unstable right now but https://github.com/artsy/force/pull/5692 should fix it. I'll rebase once it's stabilized.

File: src/v2/Apps/Auction/Components/RegistrationForm.tsx (hunk 3 above)
Comment (dleve123) [reply src/v2/Apps/Auction/Components/RegistrationForm.tsx:95]: Did you see this issue on the pure register flow? I haven't seen it before

Comment (dleve123) [reply src/v2/Apps/Auction/Components/RegistrationForm.tsx:95]: nvm! I'm wrong -- it's there

File: src/v2/Apps/Auction/Components/BidForm.tsx [hunk 4]
```diff
@@ -252,35 +252,26 @@ export const BidForm: React.FC<Props> = ({
...
-                          >
-                            {errors.agreeToTerms}
-                          </Sans>
-                        )}
                       </Box>
+
+                      {touched.agreeToTerms && errors.agreeToTerms && (
+                        <Sans mt={1} color="red100" size="2" textAlign="center">
+                          {errors.agreeToTerms}
+                        </Sans>
```
Comment (erikdstock) [inline src/v2/Apps/Auction/Components/BidForm.tsx:273]: This moved out of the box, correct?

File: src/v2/Apps/Auction/Components/BidForm.tsx [hunk 5]
```diff
@@ -289,16 +280,16 @@ export const BidForm: React.FC<Props> = ({
...
-                    width="100%"
-                    loading={isSubmitting}
-                    {...({ type: "submit" } as any)}
-                  >
-                    Confirm bid
-                  </Button>
                 </Flex>
+
+                <Button
+                  size="large"
```
Comment (erikdstock) [inline src/v2/Apps/Auction/Components/BidForm.tsx:286]: same question here, i trust all is fine but element seems to have left its container

File: src/v2/Apps/Auction/Components/BidForm.tsx (hunk 4 above)
Comment (yuki24) [reply src/v2/Apps/Auction/Components/BidForm.tsx:273]: Yes, this is intentional.
